bot/internal/bot/bot.go

```diff
@@ -153,6 +154,28 @@ func classifyChanges(e *env.Environment, files []github.PullRequestFile) env.Cha
 	return ch
 }
 
+// approverCount returns the number of required approvers for the PR by comparing the files included
+// in the PR against a set of paths that only require a single approver. 1 is returned when all of the
+// files match a single approver path, otherwise env.DefaultApproverCount is returned.
+func approverCount(singleApproverPaths []string, files []github.PullRequestFile) int {
+	if len(singleApproverPaths) == 0 || len(files) == 0 {
+		return env.DefaultApproverCount
+	}
+	for _, file := range files {
+		var match bool
+		for _, path := range singleApproverPaths {
+			if strings.HasPrefix(file.Name, path) {
```

Comment:
Currently this will match partial filenames, for example a allowed prefix of `src/dir` would also match to `src/dir2`.  Maybe not concerning, but I wanted to mention it.

Reply:
Thanks. This is by design and is the same as what we do for preferred reviewer paths.